packages/spanner/src/database.js

```diff
@@ -460,9 +460,6 @@ Database.prototype.getTransaction = function(options, callback) {
       return;
     }
 
-    options = extend({}, options);
-    delete options.readOnly;
```

Comment:
so this line was the root cause of the issue?

Reply:
More or less yes, we weren't accounting for read only transactions that did not contain timestamp options.